src/server/identity/server/api_server.go

```diff
@@ -37,10 +38,61 @@ func NewIdentityServer(env Env, public bool) identity.APIServer {
 		}()
 	}
 
-	return server
+	return server, server.envBootstrap
+}
+
+func (a *apiServer) envBootstrap(ctx context.Context) error {
+	if a.env.Config.IdentityConfig != "" {
+		a.env.Logger.Info("Started to configure identity server config via environment")
+		var config identity.IdentityServerConfig
+		if err := yaml.Unmarshal([]byte(a.env.Config.IdentityConfig), &config); err != nil {
+			return errors.Wrapf(err, "failed to unmarshal IdentityConfig with value: %s.", a.env.Config.IdentityConfig)
+		}
+		a.setIdentityServerConfig(ctx, &identity.SetIdentityServerConfigRequest{Config: &config})
+		a.env.Logger.Info("Successfully configured identity server config via environment")
+	}
+	if a.env.Config.IdentityConnectors != "" {
+		a.env.Logger.Info("Started to configure identity connectors via environment")
+		var connectors []identity.IDPConnector
+		if err := yaml.Unmarshal([]byte(a.env.Config.IdentityConnectors), &connectors); err != nil {
+			return errors.Wrapf(err, "failed to unmarshal IdentityConfig with value: %s.", a.env.Config.IdentityConfig)
+		}
+		existing, err := a.ListIDPConnectors(ctx, &identity.ListIDPConnectorsRequest{})
+		if err != nil {
+			return errors.Wrap(err, "failed to retrieve list of IDP connectors.")
+		}
+		exIds := make(map[string]struct{})
+		for _, e := range existing.Connectors {
+			exIds[e.Id] = struct{}{}
+		}
+		for _, c := range connectors {
+			if _, ok := exIds[c.Id]; ok {
+				if _, err := a.updateIDPConnector(ctx, &identity.UpdateIDPConnectorRequest{Connector: &c}); err != nil {
+					return errors.Wrapf(err, "failed to update connector with ID: %s.", c.Id)
+				}
+				delete(exIds, c.Id)
+			} else {
+				if _, err := a.createIDPConnector(ctx, &identity.CreateIDPConnectorRequest{Connector: &c}); err != nil {
+					return errors.Wrapf(err, "failed to create connector with ID: %s.", c.Id)
+				}
+			}
+		}
+		for e := range exIds {
+			a.deleteIDPConnector(ctx, &identity.DeleteIDPConnectorRequest{Id: e})
```

Comment:
If this errors, we should at least print the message as a warning.  (Avoids confusion as to why something didn't get deleted.)